proto/api/v1/auth_service.proto

```diff
@@ -20,21 +21,147 @@ message UserAttribute {
 
 message AuthStatus {
     oneof id {
-        string                user_id     = 1;
-        storage.ServiceIdentity       service_id  = 2;
+        string                  user_id         = 1;
+        storage.ServiceIdentity service_id      = 2;
     }
-    google.protobuf.Timestamp expires     = 3;
-    string                    refresh_url = 4;
+    google.protobuf.Timestamp   expires         = 3;
+    string                      refresh_url     = 4;
+    storage.AuthProvider        auth_provider   = 5;
+    storage.UserInfo            user_info       = 6;
+    repeated UserAttribute      user_attributes = 7;
+}
+
+// AuthMachineToMachineConfig determines rules for exchanging an identity token from a third party with
+// a Central access token. The M2M stands for machine to machine, as this is the intended use-case
+// for the config.
+message AuthMachineToMachineConfig {
+    // UUID of the config.
+    // Note that when adding a machine to machine config, this field should not be set.
+    string id = 1;
+
+    // The type of the auth machine to machine config.
+    // Currently supports GitHub actions or any other generic OIDC provider to use for verifying and
+    // exchanging the token.
+    enum Type {
+        GENERIC        = 0;
+        GITHUB_ACTIONS = 1;
+    }
+    Type type          = 2;
+
+    // The expiration of the token annotated as a duration.
```

Comment:
Had this at the back of my mind. `token_expiration_duration` is the desired duration of the token the caller gets back from the Exchange call, right? From my PoV, it will be worth a mention.
Something in the spirit of
```suggestion
    // Sets the life time of the token returned from the ExchangeAuthMachineToMachineToken call.
    // Must be provided in the duration format ...
```